src/core/dev_api/openvino/op/util/node_util.hpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
```

Comment:
Consider use forward declaration for `ov::Node` class. 
This header pulls a lot of other heavy headers which will cause a lot of parsing by compiler. Move this include to cpp file.